java/core/src/main/java/com/lancedb/lance/Dataset.java

```diff
@@ -41,6 +50,37 @@ public class Dataset implements Closeable {
 
   private Dataset() {}
 
+  /**
+   * Creates an empty dataset.
+   *
+   * @param path dataset uri
+   * @param schema dataset schema
+   * @param params write params
+   * @return Dataset
+   */
+  public static Dataset createEmptyDataSet(String path, Schema schema,
+      WriteParams params) {
+    // TODO(lu) move createEmptyDataset into rust lance with c.ArrowSchema
+    try (RootAllocator allocator = new RootAllocator();
+         VectorSchemaRoot root = VectorSchemaRoot.create(schema, allocator)) {
+      ByteArrayOutputStream schemaOnlyOutStream = new ByteArrayOutputStream();
+      try (ArrowStreamWriter writer = new ArrowStreamWriter(root, null,
```

Comment:
Writing data is really tedious in this way.
Anyway we can improve the java / rust API ?

Reply:
GetSchema is added, next will be adding the Spark write support.
Will work on designing/improving the write APIs after we have a clear idea of how write is called.
(Do feel it's very ugly now